resolver/resolver.cc

```diff
@@ -2059,44 +2059,38 @@ class ResolveTypeMembersAndFieldsWalk {
         }
 
         auto prior = scope.data(ctx)->findMember(ctx, uid->name);
-        if (prior.exists() && prior.isFieldOrStaticField()) {
-            auto priorField = prior.asFieldRef();
-            if (core::Types::equiv(ctx, priorField.data(ctx)->resultType, castType)) {
-                // We already have a symbol for this field, and it matches what we already saw, so we can short
-                // circuit.
-                return;
+        ENFORCE(prior.exists());
+        ENFORCE(prior.isFieldOrStaticField());
```

Comment:
I was surprised that when I tested this, it doesn't trip for this example:

```ruby
# typed: true

class A
  X = 1
  X = type_member
end

class B
  X = type_member
  X = 1
end
```

Do you understand how the mangling works here? I guess that maybe we always mangle the type member symbol, and deal with that elsewhere?

Reply:
Type members (and non-class variable static fields generally) are dealt with in a different code path:

https://github.com/sorbet/sorbet/blob/d315899bda4c76868181c98b748cc72a82ee0ef5/resolver/resolver.cc#L2915-L2976

`handleFieldDeclaration` pushes a `ResolveFieldItem` job for instance variables and class variable assignments.  General static field resolution is handled elsewhere.